src/System Application/App/Password/src/PasswordDialogImpl.Codeunit.al

```diff
@@ -73,6 +75,7 @@ codeunit 9811 "Password Dialog Impl."
         PasswordDialog: Page "Password Dialog";
     begin
         PasswordDialog.EnableChangePassword();
+        PasswordDialog.SetOldPasswordToCompareSecretValue(OldPassword);
```

Comment:
The OldPassword parameter is a bit confusing now.. So as an input it's the old password that should be compared to, as an output it's the old password the user typed. Now these are not necessarily equal, if the OldPassword input was empty, then the OldPassword output is not necessarily empty. Of course if OldPassword input has a value, then OldPassword output are equal.

Reply:
Exactly, they can have differences and it's all because the original function had a very strange design where the OldPassword was just what the user entered in the dialog box, all the checks were kind of out of brackets. This in turn created confusion about how to use the function at all. 

https://github.com/microsoft/BCApps/issues/328#issuecomment-1983618268

To support backwards compatibility, when passing a blank to OldPassword (and this function was very often used this way) it will work as it did before the change. But only when OldPassword is filled in will the function actually perform a comparison. In addition, checking that the old password does not match the new one makes sense only when the old password has been sent. Otherwise there is nothing to check.

So the logic is as follows
Input OldPassword = '' works as before, we just return the entered password, no checks.
Input OldPassword <> '' if we do pass OldPassword it means that we assume a specific OldPassword and if it doesn't match the password the user fills in, an error is inevitable and necessary. Only in this case there is any chance to compare if OldPassword doesn't match NewPassword.

I'd say more confusing was how the function worked previously